### DISCUSSION TITLE
[Doc] Add introduction to Commanded's concepts in "Getting started" guide

### DISCUSSION THREAD

**Discussion by slashdotdash** [commanded/commanded#593] [General]
To help onboard new users it would be useful to describe the main building blocks required to start using Commanded and configure an event store.

- `Commanded.Application` - explain what it is, and that it differs from an OTP application.
- Event store - adapter configuration and event store definition (e.g. define `EventStore` module).

**Comment by J3RN** [comment]
Related to the first point, I think it might be appropriate to establish a convention for the file in which to use `Commanded.Application`. [The current Application documentation](https://hexdocs.pm/commanded/1.3.1/application.html) uses `MyApp.Application`, but almost every Elixir application already has such a file that configures it's OTP application.

Some real-world examples are [Bonfire uses `Bonfire.EventApp`](https://github.com/qhwa/bonfire/blob/4a368d6d5300539399dcaff167ac69e3165c2bff/lib/bonfire/event_app.ex) and [@vasspilka's Bank example creates a `Bank.Core` namespace with the Commanded application at `Bank.Core.Application`](https://github.com/vasspilka/bank/blob/6da9b9e45cd4880d0e421e2702de3ea33c27ebf1/lib/bank/core/application.ex).

**Comment by byu** [comment]
As a brand new user, this was definitely something that confused me. Upvote for some clarifying "best practice" documentation about how to structure Commanded Applications in both fresh projects (ala `mix new --sup`) and within phoenix projects.
